### PR TITLE
add dhcp_hostname config for esp32

### PR DESCRIPTION
**Pull request by andrey-hinkov**

### CODE REVIEW

File: src/esp32/esp32_eth.c [hunk 1]
```diff
@@ -126,6 +137,12 @@ bool mgos_ethernet_init(void) {
     return false;
   }
 
+  ret = esp32_set_host_name(mgos_sys_config_get_eth());
```
Comment (cpq) [inline src/esp32/esp32_eth.c]: I would set host name only if it is explicitely set in the configuration:

```c
if (mgos_sys_config_get_eth_dhcp_hostname() != NULL &&
    esp_netif_set_hostname(esp_netif_get_handle_from_ifkey("ETH_DEF"), 
      mgos_sys_config_get_eth_dhcp_hostname()) != ESP_OK) {
  LOG(LL_ERROR, ("ETH: Failed to set host name"));
  goto out;
}
```

Comment (andrey-hinkov) [reply src/esp32/esp32_eth.c]: Should be fixed now, thanks for the advice. https://github.com/mongoose-os-libs/ethernet/compare/master...ALLTERCO:master

Comment (rojer) [reply src/esp32/esp32_eth.c]: i think the function can be inlined since it's only used once. you already have the netif handle in `eth_if` here

**Comment by kzyapkov** [comment]
Strangely, pushing new versions of the commit doesn't reflect this pull request. Let's close this, I'll open a fresh one from a branch in our fork.

**Comment by cpq** [comment]
Closing this, as https://github.com/mongoose-os-libs/ethernet/pull/6 was pushed.

**Pull request closed without merging**